internal/provider/provider.go

```diff
@@ -100,8 +102,15 @@ func (p *ProviderFramework) Schema(ctx context.Context, req provider.SchemaReque
 				NestedObject: schema.NestedBlockObject{
 					Attributes: map[string]schema.Attribute{
 						"token_file": schema.StringAttribute{
-							Required:    true,
-							Description: "The path to a file containing a JWT token retrieved from an OpenID Connect (OIDC) or OAuth2 provider.",
+							Optional:    true,
+							Description: "The path to a file containing a JWT token retrieved from an OpenID Connect (OIDC) or OAuth2 provider. Exactly one of `token_file` or `token` must be set.",
+							Validators: []validator.String{
+								stringvalidator.LengthAtLeast(1),
```

Comment:
Could we use `stringvalidator.ConflictsWith` or something similar to have TF validate only one is set?

Reply:
Nice, didn't know they had such specific validators. I did end up using `ExactlyOneOf` instead of just `ConflictsWith` - I think that has the same effect but with an additional validation that one must be set.